### PR TITLE
P1 345 woocommerce snippet in google preview

### PR DESCRIPTION
**Pull request by marijnyoast**
## Context

* We want our Google Preview to reflect how the search result for a product looks in Google. So when the necessary data for Product schema is available, it should show up in the Google Preview, like this:

image

image

## Summary

This PR can be summarized in the following changelog entry:

* When used together with the latest version of Wordpress SEO or Wordpress SEO Premium, the Google Preview for products now shows the rating, number of reviews, price and availability like in Google.

## Relevant technical choices:

* For the sake of simplicity we decided to directly dispatch to the WPSEO store.
* Since the needed data is not yet available in WooCommerce's stores, this data is static for now.

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* Test together with the related branches on Free and the Monorepo, of for QA with the latest RC/Beta of Free/Premium 16.2
* Edit a product and save it
* Refresh the page
* Repeat this process with different data added and:
** Verify that when a review has been entered the rating and number of reviews will show up in the Google Preview
** Verify that when a price has been entered the price shows op in the Google Preview
** Verify that when the availability is set or changed it shows up in the Google Preview
* Please check the mobile as well as the desktop view
* Please note that a change is only reflected in the preview after a save and refresh

### Test instructions for QA when the code is in the RC

* [X] QA should use the same steps as above.

QA can test this PR by following these steps:

*

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes

* [ ] This PR changes the UI in the plugin. I have added the 'UI change' label to this PR.

## Documentation

* [ ] I have written documentation for this change.

## Quality assurance

* [X] I have tested this code to the best of my abilities
* [ ] I have added unittests to verify the code works as intended

Fixes P1-345

### CODE REVIEW

**Review by igorschoester — CHANGES_REQUESTED** [review]
CR 🏗️

File: classes/woocommerce-seo.php [hunk 1]
```diff
@@ -76,7 +76,6 @@ protected function initialize() {
 		else {
 			// Initialize schema & OpenGraph.
 			add_action( 'init', [ $this, 'initialize_opengraph' ] );
-			add_action( 'init', [ $this, 'initialize_schema' ] );
```
[the remark this answers is not in the document]
Comment (marijnyoast) [reply classes/woocommerce-seo.php]: good spot, it's back!

File: js/src/yoastseo-woo-plugin.js [hunk 2]
```diff
@@ -67,6 +70,19 @@ class YoastWooCommercePlugin {
...
+	/**
+	 * Dispatches the product data from window.wpseoWooL10n to the Yoast SEO editor store.
+	 *
+	 * @returns {void}
+	 */
+	dispatchGooglePreviewData() {
+		const googlePreviewData = window.wpseoWooL10n.wooGooglePreviewData;
+
+		if ( dispatch && googlePreviewData  ) {
+			dispatch( "yoast-seo/editor" ).setShoppingData( googlePreviewData );
```
[the remark this answers is not in the document]
Comment (marijnyoast) [reply js/src/yoastseo-woo-plugin.js:81]: It seems to work without this faulty check indeed

File: js/src/yoastseo-woo-plugin.js [hunk 3]
```diff
@@ -76,9 +76,8 @@ class YoastWooCommercePlugin {
 	 * @returns {void}
 	 */
 	dispatchGooglePreviewData() {
-		const googlePreviewData = window.wpseoWooL10n.wooGooglePreviewData;
-
-		if ( dispatch && googlePreviewData  ) {
+		if ( window.wpseoWooL10n.wooGooglePreviewData ) {
```
Comment (igorschoester) [inline js/src/yoastseo-woo-plugin.js:79]: I don't think this check is needed. You always add this key to the L10n. The L10n is always there due to the localize_script in PHP.
If you really want, you could add an empty check and then not dispatch if there is no data to add. But looking at what currently is in `wooGooglePreviewData`, this seems to always have something.

**Review by igorschoester — COMMENTED** [review]
CR 🏗️

**Review by igorschoester — APPROVED** [review]
CR 👍

**Comment by JoseeWouters** [comment]
Acceptance 👍

**Pull request merged**